core/common/src/test/java/alluxio/EmbeddedChannels.java

```diff
@@ -80,5 +103,352 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
         recordException(cause);
       }
     }
+
+    /**
+     * The channel pipeline that ensures a {@link LastInboundHandler} instance is always in the
+     * end. This is a wrapper of the given pipeline instance with modified {@link #addLast} which
+     * ensures the invariance.
+     */
+    public class EmbeddedChannelPipeline implements ChannelPipeline {
+      private final ChannelPipeline mDefaultChannelPipeline;
```

Comment:
NIT: can this just be called `mPipeline`?

Reply:
fixed